starter/app/src/main/java/com/udacity/project4/locationreminders/savereminder/SaveReminderFragment.kt

```diff
@@ -1,61 +1,274 @@
+        intent.action = ACTION_GEOFENCE_EVENT
+        intent.putExtra(GEO_FENCE_ID, geofenceData.id)
+        intent.putExtra(GEO_FENCE_LAT, geofenceData.latLong.latitude)
+        intent.putExtra(GEO_FENCE_LNG, geofenceData.latLong.longitude)
+        // Use FLAG_UPDATE_CURRENT so that you get the same pending intent back when calling
+        // addGeoFences() and removeGeoFences().
+        var intentFlagTypeUpdateCurrent = PendingIntent.FLAG_UPDATE_CURRENT
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            intentFlagTypeUpdateCurrent = PendingIntent.FLAG_IMMUTABLE
+        }
+        PendingIntent.getBroadcast(requireContext(), 0, intent, intentFlagTypeUpdateCurrent)
+    }
 
-        setDisplayHomeAsUpEnabled(true)
+    //--------------------------------------------------
+    // Lifecycle Methods
+    //--------------------------------------------------
 
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View {
+        Log.d(TAG, "SaveReminderFragment.onCreateView().")
+
+        val layoutId = R.layout.fragment_save_reminder
+        binding = DataBindingUtil.inflate(inflater, layoutId, container, false)
         binding.viewModel = _viewModel
+        init()
 
         return binding.root
     }
 
+    private fun init() {
+        Log.d(TAG, "SaveReminderFragment.init().")
+        setDisplayHomeAsUpEnabled(true)
+        parent = requireActivity()
+        parent.onBackPressedDispatcher.addCallback(viewLifecycleOwner, backPressHandler)
+
+        getNavigationResult(ARGUMENTS)?.observe(viewLifecycleOwner) { result ->
+            val triple = result as Triple<String, Double, Double>
+            _viewModel.reminderSelectedLocationStr.value = triple.component1()
+            _viewModel.latitude.value = triple.component2()
+            _viewModel.longitude.value = triple.component3()
+        }
+
+        val menuHost: MenuHost = parent
+        addMenu(menuHost)
+
+        geofencingClient = LocationServices.getGeofencingClient(parent)
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        Log.d(TAG, "SaveReminderFragment.onViewCreated().")
+
         binding.lifecycleOwner = this
         binding.selectLocation.setOnClickListener {
-            //            Navigate to another fragment to get the user location
-            _viewModel.navigationCommand.value =
-                NavigationCommand.To(SaveReminderFragmentDirections.actionSaveReminderFragmentToSelectLocationFragment())
+            // Navigate to another fragment to get the user location
+            val directions = SaveReminderFragmentDirections
+                .actionSaveReminderFragmentToSelectLocationFragment()
+            _viewModel.navigationCommand.value = NavigationCommand.To(directions)
         }
 
+        // Use the user entered reminder details to:
```

Comment:
@graffiti75 I can see this a TODO, so could you add the word TODO?

Reply:
New PR created: https://github.com/udacity/nd940-android-kotlin-c4-starter/pull/14